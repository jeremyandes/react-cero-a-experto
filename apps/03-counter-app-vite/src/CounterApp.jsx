import PropTypes from 'prop-types'

export const CounterApp = ({ value }) => {
    const clickEvent = (event) => console.log(event);

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{value}</h2>
            <button onClick={clickEvent}>
                +1
            </button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}