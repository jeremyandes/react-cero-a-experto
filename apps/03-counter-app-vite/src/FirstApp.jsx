import PropTypes from 'prop-types';

export const FirstApp = ({ title, subtitle }) => {

    return (
        <>
            <h1 data-testid='test-title'>{title}</h1>
            {/* <code>{JSON.stringify(newMessage)}</code> */}
            <h4>{subtitle}</h4>
            <h4>{subtitle}</h4>
            <h4>{subtitle}</h4>
        </>
    );

}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.number.isRequired,
}

FirstApp.defaultProps = {
    // title: 'No hay t√≠tulo',
    subtitle: 123,
}