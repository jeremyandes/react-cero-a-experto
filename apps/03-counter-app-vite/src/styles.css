html,
body {
    background-color: #21232A;
    color: white;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 1.25rem;
}

body {
    display: flex;
    justify-content: center;
    align-items: center;
    text-align: center;
    flex-direction: column;
    width: 100vw;
    height: 100vh;
}

button {
    padding: 15px;
    font-size: 1.25rem;
}