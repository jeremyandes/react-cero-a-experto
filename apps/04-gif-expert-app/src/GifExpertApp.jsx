import { useState } from 'react';
import { AddCategory } from './components/AddCategory';

export const GifExpertApp = () => {
    const [categories, setCategories] = useState(['Michael Jordan', 'Curry']);

    const onAddCategory = () => {
        setCategories([...categories, 'Facu Campazzo']);
        // setCategories(cats => [ ...cats, 'Facu Campazzo' ]);
    }

    return (
        <>
            {/* Titulo */}
            <h1>GifExpertApp</h1>

            {/* Input */}
            <AddCategory />

            {/* Gifs */}
            <button type='button' onClick={onAddCategory}>Agregar</button>
            <ol>
                {
                    categories.map(category => {
                        return <li key={category}>{category}</li>
                    })
                }
            </ol>
            {/* Item */}
            {/* EndGifs */}
        </>
    )
}
