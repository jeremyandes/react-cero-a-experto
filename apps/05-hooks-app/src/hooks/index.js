export * from './useForm';
export * from './useFetch';
export * from './useTodos';
export * from './useCounter';