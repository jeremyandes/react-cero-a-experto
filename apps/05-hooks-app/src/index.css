* {
    margin: 0;
    padding: 0;
    bottom: 0;
    box-sizing: border-box;
}

html {
    scroll-behavior: smooth;
}

body,
#root {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    width: 100%;
    min-height: 100vh;
}

#root {
    padding: 25px;
}

button {
    margin: 5px;
}