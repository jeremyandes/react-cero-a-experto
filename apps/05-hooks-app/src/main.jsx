import React from 'react'
import ReactDOM from 'react-dom/client'
import { CounterApp } from './01-useState/CounterApp'
import { CounterCustomHook } from './01-useState/CounterCustomHook'
import { FormCustomHook } from './02-useEffect/FormCustomHook'
import { SimpleForm } from './02-useEffect/SimpleForm'
import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks'
import { FocusScreen } from './04-useRef/FocusScreen'
import { HooksApp } from './HooksApp'
import './index.css'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/* <HooksApp /> */}
    {/* <CounterApp /> */}
    {/* <CounterCustomHook /> */}
    {/* <SimpleForm /> */}
    {/* <FormCustomHook /> */}
    {/* <MultipleCustomHooks /> */}
    <FocusScreen />
  </React.StrictMode>
)
