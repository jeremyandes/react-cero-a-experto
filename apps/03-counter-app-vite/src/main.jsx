import React from 'react';
import ReactDOM from 'react-dom/client';

// import { App } from './App';
import { FirstApp } from './FirstApp';
import { CounterApp } from './CounterApp';

import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp value={25} />
        <FirstApp title='hola, soy Jeremy' />
    </React.StrictMode>
);