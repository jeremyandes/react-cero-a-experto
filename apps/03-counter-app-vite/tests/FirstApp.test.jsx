import { render } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Pruebas en <FirstApp />', () => {

    // test('Debe hacer match con el snapshot', () => {

    //     const title = 'Hola, soy Jeremy';
    //     const { container } = render(<FirstApp title={title} />);
    //     expect(container).toMatchSnapshot();

    // })

    test('Debe mostrar el tÃ­tulo en un H1', () => {

        const title = 'Jeremy';
        const { container, getByText, getByTestId } = render(<FirstApp title={title} />);
        expect(getByText(title)).toBeTruthy();

        // const h1 = container.querySelector('h1');
        // expect(h1.innerHTML).toContain(title);

        expect(getByTestId('test-title').innerHTML).toContain(title);
    })

    test('Debe mostrar el titulo enviado por props', () => {

        const title = 'Jeremy';
        const subtitle = 2525;
        const { getAllByText } = render(
            <FirstApp
                title={title}
                subtitle={subtitle}
            />
        );
        expect(getAllByText(subtitle).length).toBe(3);
    })

})