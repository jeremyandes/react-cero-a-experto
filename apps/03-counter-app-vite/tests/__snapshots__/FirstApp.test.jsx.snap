// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, soy Jeremy
  </h1>
  <h4>
    123
  </h4>
</div>
`;
